core/src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -32,268 +31,545 @@
+                            throw new ParsingException(parser.getTokenLocation(),
+                                "unexpected value [" + (token.isValue() ? parser.text() : token) + "], expected [<option name>]");
+                        }
+                    }
+                } else {
+                    throw new ParsingException(parser.getTokenLocation(),
+                        "unexpected value [" + (token.isValue() ? parser.text() : token) + "], expected [<options>]");
+                }
+            } else if (matcher.match(name, ScriptField.PARAMS)) {
+                if (token == Token.START_OBJECT) {
+                    params = parser.map();
+                } else {
+                    throw new ParsingException(parser.getTokenLocation(),
+                        "unexpected value [" + (token.isValue() ? parser.text() : token) + "], expected [<params>]");
+                }
+            } else {
+                throw new ParsingException(parser.getTokenLocation(),
+                    "unexpected field [" + (name == null ? parser.currentToken() : name) + "], " + "expected [" +
+                        ScriptType.INLINE.getParseField().getPreferredName() + ", " +
+                        ScriptType.STORED.getParseField().getPreferredName() + ", " +
+                        ScriptType.FILE.getParseField().getPreferredName() + ", " +
+                        ScriptField.LANG.getPreferredName() + ", " +
+                        ScriptField.PARAMS.getPreferredName() +
+                        "]");
+            }
+        }
+
+        if (type == null) {
+            throw new IllegalArgumentException("missing required parameter for parsing script, expected valid script type [" +
+                ScriptType.INLINE.getName() + "] or [" + ScriptType.STORED.getName() + "] or [" + ScriptType.FILE.getName() + "]");
+        }
+
+        if (idOrCode == null) {
+            throw new IllegalArgumentException("missing required parameter for parsing script, expected [<id>] or [<code>]");
+        }
+
+        if (params == null) {
+            params = new HashMap<>();
+        }
+
+        return new Script(type, lang, idOrCode, options, params);
+    }
+
     @SuppressWarnings("unchecked")
-    public Script(String script, ScriptType type, String lang, @Nullable Map<String, ?> params,
-                  @Nullable XContentType  contentType) {
-        if (contentType != null && type != ScriptType.INLINE) {
-            throw new IllegalArgumentException("The parameter contentType only makes sense for inline scripts");
+    public static Script readFrom(StreamInput in) throws IOException {
+        // Version 6.0 requires all Script members to be non-null and supports the potential
+        // for more options than just XContentType.  Reorders the read in contents to be in
+        // same order as the constructor.
+        if (Version.smallest(in.getVersion(), Version.V_6_0_0_alpha1) == Version.V_6_0_0_alpha1) {
```

Comment:
I honestly completely missed that this method was in there after I saw the smaller.  Fixed.